pkg/virt-controller/watch/vm.go

```diff
@@ -1151,16 +1138,34 @@ func (c *VMController) applyFlavorToVmi(vm *virtv1.VirtualMachine, vmi *virtv1.V
 }
 
 func hasStartPausedRequest(vm *virtv1.VirtualMachine) bool {
-	if len(vm.Status.StateChangeRequests) != 0 {
-		stateChange := vm.Status.StateChangeRequests[0]
-		if stateChange.Action == virtv1.StartRequest {
-			paused, hasPaused := stateChange.Data[virtv1.StartRequestDataPausedKey]
-			if hasPaused && paused == virtv1.StartRequestDataPausedTrue {
-				return true
-			}
-		}
+	if len(vm.Status.StateChangeRequests) == 0 {
+		return false
 	}
-	return false
+
+	stateChange := vm.Status.StateChangeRequests[0]
+	pausedValue, hasPaused := stateChange.Data[virtv1.StartRequestDataPausedKey]
+	return stateChange.Action != virtv1.StartRequest &&
+		hasPaused && pausedValue == virtv1.StartRequestDataPausedTrue
+}
+
+func hasStartRequest(vm *virtv1.VirtualMachine) bool {
+	if len(vm.Status.StateChangeRequests) == 0 {
+		return false
+	}
+
+	stateChange := vm.Status.StateChangeRequests[0]
+	return stateChange.Action != virtv1.StartRequest
```

Comment:
should this be `return stateChange.Action != virtv1.StartRequest` ? or am i reading this incorrectly. It seems like the way it's written that `hasStartRequest` returns true if the action isn't a start request.

Reply:
Yep I got this wrong. Also in the above `hasStartPausedRequest`. This also explains the many sig-compute lane test failures that I haven't yet had the time to debug.

Reply:
Fixed.